Automattic/newspack-newsletters
# feat: override publish button

### All Submissions:


### Changes proposed in this Pull Request:

Overrides the Gutenberg publish button. The approach uses React Portals. The `Editor` component creates a DOM element and inserts in front of the publish button. This element is used as a portal to be filled with the custom publish button. The default publish button is hidden by CSS. The custom publish button's label reflects its purpose: "Send Newsletter" | "Sending Newsletter" | "Sent". The button is not enabled until the newsletter is ready to send, and consequently the pre-publish checklist has been removed. Currently there is no support for scheduling publishing. 

### How to test the changes in this Pull Request:

1. Create a Newsletter.
2. Verify publish button has text "Send Newsletter" and is disabled.
3. Select a list, input a sender name and email. Verify the publish button is now enabled.
4. Send the Newsletter. Verify the publish button is now disabled and reads "Sent."

### Other information:
